### PR TITLE
eahw-2251-add-maven-checkstyle: add the maven checkstyle plugin

### PR DESCRIPTION
**Pull request by JWeston-HO**
Adding the maven checkstyle plugin to Jparest. Using google checkstyle. 

Javadoc is suppressed.

### CODE REVIEW

File: jparest/pom.xml.versionsBackup [hunk 1]
```diff
@@ -0,0 +1,104 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (stephenkaye-ho) [inline jparest/pom.xml.versionsBackup:1]: This should be removed

File: checkstyle_suppressions.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0"?>
+
+<!DOCTYPE suppressions PUBLIC
+        "-//Puppy Crawl//DTD Suppressions 1.2//EN"
+        "http://www.puppycrawl.com/dtds/suppressions_1_2.dtd">
+
+<suppressions>
+    <suppress checks="Javadoc" files="."/>
```
Comment (stephenkaye-ho) [inline checkstyle_suppressions.xml:8]: Is this best practice to disable Javadoc warnings?

**Review by stephenkaye-ho — DISMISSED** [review]
The backup file should be removed

File: checkstyle_suppressions.xml (hunk 2 above)
Comment (JWeston-HO) [reply checkstyle_suppressions.xml:8]: Happy to remove. But we would have to add what is document in the below link to all our java files. 
https://checkstyle.sourceforge.io/config_javadoc.html

File: jparest/pom.xml.versionsBackup (hunk 1 above)
Comment (JWeston-HO) [reply jparest/pom.xml.versionsBackup]: removed

**Review by daleelR — APPROVED** [review]

**Review by daleelR — APPROVED** [review]

**Review by laurieboyes — APPROVED** [review]

**Review by daleelR — APPROVED** [review]

**Review by laurieboyes — APPROVED** [review]

**Pull request merged**